src/ec2/model.ts

```diff
@@ -113,19 +128,29 @@ export class Ec2ConnectionManager {
         const hasPermission = await this.hasProperPermissions(IamRole!.Arn)
 
         if (!hasPermission) {
-            const message = `Ensure an IAM role with the proper permissions is attached to the instance. Found attached role: ${
-                IamRole!.Arn
-            }`
-            this.throwConnectionError(message, selection, {
-                code: 'EC2SSMPermission',
-                documentationUri: this.policyDocumentationUri,
-            })
+            const policiesAdded = await promptToAddPolicies(this.iamClient, IamRole!.Arn!)
+            if (!policiesAdded) {
+                const message = `Did not add permissions. Ensure an IAM role with the proper permissions is attached to the instance. Found attached role: ${
+                    IamRole!.Arn
+                }`
+                this.throwConnectionError(message, selection, {
+                    code: 'EC2SSMPermission',
+                    documentationUri: this.policyDocumentationUri,
+                })
+            } else {
+                const timeout = new Timeout(policyAttachDelay)
+
+                function delay(ms: number) {
+                    return new Promise(resolve => setTimeout(resolve, ms))
+                }
+                await showMessageWithCancel(`Adding Inline Policy to ${IamRole!.Arn}`, timeout)
```

Comment:
`showMessageWithCancel` has a `showAfterMs` parameter. 
```suggestion
                await showMessageWithCancel(`Adding Inline Policy to ${IamRole!.Arn}`, timeout, policyAttachDelay)
```